plugins/devices/marytts/src/main/java/com/freedomotic/plugins/marytts/MaryTTS.java

```diff
@@ -0,0 +1,175 @@
+/**
+ *
+ * Copyright (c) 2009-2015 Freedomotic team http://freedomotic.com
+ * 
+* This file is part of Freedomotic
+ * 
+* This Program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2, or (at your option) any later version.
+ * 
+* This Program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ * 
+* You should have received a copy of the GNU General Public License along with
+ * Freedomotic; see the file COPYING. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @author Mauro Cicolella
+ */
+
+package com.freedomotic.plugins.marytts;
+
+import com.freedomotic.api.EventTemplate;
+import com.freedomotic.api.Protocol;
+import com.freedomotic.app.Freedomotic;
+import com.freedomotic.exceptions.PluginStartupException;
+import com.freedomotic.exceptions.UnableToExecuteException;
+import com.freedomotic.reactions.Command;
+import com.freedomotic.util.Info;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sound.sampled.AudioInputStream;
+
+import marytts.LocalMaryInterface;
+import marytts.MaryInterface;
+import marytts.exceptions.MaryConfigurationException;
+import marytts.exceptions.SynthesisException;
+import marytts.modules.synthesis.Voice;
+import marytts.util.data.audio.AudioPlayer;
+
+public class MaryTTS extends Protocol {
+
+    private static final Logger LOG = Logger.getLogger(MaryTTS.class.getName());
+    private MaryInterface marytts;
+    private Voice defaultVoice;
+    private Voice voice;
+    private AudioInputStream audio;
+    private String VOICE_JAR_FILE = configuration.getProperty("voice-jar-file");
+    private String VOICE = configuration.getProperty("voice");
+
+    public MaryTTS() {
+        super("MaryTTS", "/marytts/marytts-manifest.xml");
+    }
+
+    @Override
+    protected void onStart() throws PluginStartupException {
+        try {
+            //add voices folder to classpath
+            addPath(Info.PATHS.PATH_DEVICES_FOLDER + System.getProperty("file.separator") + "marytts/lib/" + VOICE_JAR_FILE);
```

Comment:
lib folder should be already on the plugin classpath, is it now working?

this will not work on all OS because you use file.separator and then you hardcode the file separator in the next piece of string "/". Pass a File instance used to build the path